### ISSUE TITLE
Need to implement WAIT , tips ? 

### ISSUE THREAD

**Issue by bklooste** [StackExchange/StackExchange.Redis#1474]
Have read the other thread but i need this anyway (not immediately but soon), don't need other blocking operations just WAIT 

My options are 
-  new ConnectionMultiplexer  possibly in a separate domain , i see people have issues but try to disable thread steal and set higher sync time outs.  Look at stalling code in stack exchange
-  Try get a connection from the lib and just send the commands. 
-  fork stack exchange , make a change there.  

Just after some tips / suggestions of the best approach.

**Comment by mgravell** [comment]
We know that `WAIT` is a problem. We have investigated some alternative internal implementations that will enable `WAIT` and other blocking operations - essentially, switching to a connection pool internally. This also resolves a number of other outstanding problems. The prototype was very successful - but we have not yet merged this work, because: it takes time, and this isn't our day jobs. I have a few other plates that I'm spinning at the moment, but this is very high on my list.

Blocking operations are anathema to the current implementation; there are no supported workarounds; you can hack them in via `Execute` etc, but: this is **not** advisory, and no support will be offered to make it work, because it *cannot* work reliably.

My advice: wait

**Comment by bklooste** [comment]
Thank you much appreciated.  I presume this will be a major release with a long cycle ?

**Comment by mgravell** [comment]
No, I was going to do this as 2.2, and the proof-of-concept is complete - it just needs integrating and testing, but: this isn't my only project and isn't my day job, so: it is tied on my availability. It is something I *really* want to get in, because it simplifies the IO path *hugely*, which should resolve a ton a network pain points.

**Comment by MikaelElkiaer** [comment]
Does this mean that there are currently no means of ensuring replication?

I am planning to set up master-replicant with sentinels. I suspect I have some data that I need to ensure is the same no matter which node you reach. It seems this is only possible with `WAIT`.

Is there anything going against an approach using a lua script to perform synchronous operations - i. e. using `WAIT`.

**Comment by mgravell** [comment]
That Lua script will block your entire server, and could even get forcibly
terminated. I don't recommend that.

**Comment by MikaelElkiaer** [comment]
> That Lua script will block your entire server, and could even get forcibly
terminated. I don't recommend that.

Ok, if that is more or less ruled out, then what else is there left to do?
Even when sticking with `DemandMaster`, a failover will not be consistent either.
Is there then only going full cluster and manually write all masters, while keeping `DemandMaster`?

Is this a problem of client and/or server blocking? I'd assume that server blocking always would be possible despite any limitations in the client - what am I missing here?

**Comment by MikaelElkiaer** [comment]
@mgravell Sorry for bumping, but I was wondering if you could come with some clarification regarding the last bit of my previous post. Wouldn't it be possible to do `WAIT` within the context of a `MULTI` - even in a multiplexer scenario?

**Comment by mgravell** [comment]
Honestly, until I get the time to implement pooled connections: I am not, in any scenario, going to advocate anything that introduces a blockage at the server, no matter how transient. I can see many ways it go go badly. Are there scenarios where it might work acceptably under modest load? Possibly, but I'm not going to try and think through what the limits are.

**Comment by MikaelElkiaer** [comment]
> Honestly, until I get the time to implement pooled connections: I am not, in any scenario, going to advocate anything that introduces a blockage at the server, no matter how transient. I can see many ways it go go badly. Are there scenarios where it might work acceptably under modest load? Possibly, but I'm not going to try and think through what the limits are.

Fair enough. Thanks for the reply, will keep an eye out for connection pooling.

**Comment by aradalvand** [comment]
> The prototype was very successful - but we have not yet merged this work

@NickCraver It's been 6 years... Still can't run blocking operations with `StackExchange.Redis`. This is disappointing.
